sdk/src/store.rs

```diff
@@ -1337,12 +1337,15 @@ impl Store {
 
     // generate a list of AssetHashes based on the location of objects in the stream
     #[cfg(feature = "sign")]
-    fn generate_data_hashes_for_stream(
-        stream: &mut dyn CAIReadWrite,
+    fn generate_data_hashes_for_stream<R>(
+        stream: &mut R,
```

Comment:
This seems like a good change. :+1:

Reply:
Yeah, now we can accept anything that has a Read + Seek trait.